docs/modules/datasources/pages/flow_datasource.adoc

```diff
@@ -1,209 +1,276 @@
-```
-interfacesOnExporterNodeWithFlows(123)
-```
+| asTableSummary
+| Generate a tabular summary instead of time series data.
 
-To query a specific node by foreign source and foreign id, you can use an expression similar to:
-```
-interfacesOnExporterNodeWithFlows(FS:FID)
-```
+| combineIngressEgress
+| Combine ingress and egress values.
 
-=== dscpOnExporterNodeAndInterface(<nodeCriteria>,<interface>,<from>,<to>)
+| onlyIngress
+| Display only ingress traffic.
 
-The `dscpOnExporterNodeAndInterface()` query will return a list of DSCP values which are used by one or more flows on the given node and interface
-and given time range. The query returns selection options for the found DSCP values and ip precedence groups that cover these values.
-Labels of selection options for DSCP codes are symbolic names for these DSCP codes followed by the corresponding number (if applicable).
+| onlyEgress
+| Display only egress traffic.
 
-The available arguments are as follows:
-[options="header, autowidth"]
-|===
-| Name                 | Default     | Description
-| nodeCriteria         | `(none)`    | Node criteria (Node ID, or Node Foreign Source and Node Foreign ID) of the node on which to enumerate.
-| interface            | `(none)`    | Interface Index of the SNMP interface to enumerate.
-| from                 | `(none)`    | Start time in epoch milliseconds
-| to                   | `(none)`    | End time in epoch milliseconds
+| withGroupByInterval
+| Changes the resolution of the returned datapoints grouping by the given time interval (`10s`, `5m`, `1h`, and so on).
+
+| withPrefix
+| Specifies a prefix for graph and summary labels.
+
+| withSuffix
+| Specifies a suffix for graph and summary labels.
+
+| nanToZero
+| Convert all `NaN` values to zero (`0`) so they can be taken into consideration by reduce expressions (`sum`, `max`).
 |===
 
-Assuming that the `$node` variable is populated with a value from the `exporterNodesWithFlows` query and that the `$interface` variable is populated with a value from the `interfacesOnExporterNodeWithFlows($node)` query, you can use the following expression:
-```
-dscpOnExporterNodeAndInterface($node, $interface, $__from, $__to)
-```
+[[ds-flow-template]]
+== Template queries
 
-=== applications(<limit>)
+=== locations()
 
-The `aplications()` query returns a list of all available applications.
-This function is intended for use inside the Filter function `withApplications()` as a template variable if needed.
+The `locations()` query returns a list of available node locations.
+You can use this inside a `exporterNodesWithFlows()` query to filter the number of nodes displayed by location.
 
-Currently this function accepts an optional number of records which ensures that only a limited number or records will be retreieved.
+.Example expressions using `locations()`
+[source,]
+----
+exporterNodesWithFlows(location='Default')
+exporterNodesWithFlows(location='$location')
+----
```

Comment:
See note on performance datasource - technically these examples are not showing the use of `locations()`
```suggestion
exporterNodesWithFlows(location='$location') <1>
----
<1> Assumes a `$location` variable based on a `locations()` query.
```

Reply:
Could you provide an example of a `locations()` query, or is it one that accepts no arguments and simply returns a result?

Reply:
locations() is meant to be used as a parameter 
When a user selects a location from the filter, the selected location can be used in directly in the query editor or in another template variable.

![Screenshot 2023-02-21 at 9 32 22 AM](https://user-images.githubusercontent.com/3297292/220375192-6b44cc58-02eb-4843-a2d3-5d8bcf72cdfd.png)

![Screenshot 2023-02-21 at 9 32 03 AM](https://user-images.githubusercontent.com/3297292/220375253-959911c8-35cd-4b14-8c52-7443fc54b192.png)

Reply:
Yeah, this is a better example that shows using two variables to feed `locations()` into a `nodeFilter()`

Reply:
```suggestion
nodeFilter(location='$location') <1>
----
<1> Assumes that the `$location` variable is populated with a value from a `locations()` query.
```